main.py

```diff
@@ -130,13 +137,23 @@ def run_experiment(dataset, compressor, top_k_accuracy, compressors_per_class, s
                     size_results.append(size_result)
 
     # todo add save in proper tabular format
+
+    #so basically i create list comprehension of all keys in first dict
```

Comment:
let's remove this comment
if you want to make things clearer, I suggest: 

```
accuracy_headers = [x for x in results[0].keys()]
write_csv(os.path.join(os.getcwd(), 'accuracy_results.csv'), results, accuracy_headers)

...
speed_result_headers = [x for x in speed_results[0].keys()]
write_csv(os.path.join(os.getcwd(), 'speed_results.csv'), speed_results, speed_result_headers)

other option is to encapsulate this in the `write_csv` function

Reply:
I encapsulated all of it into the function. It only takes the two arguments now, and handles the path stuff for the first argument within the function.